packages/server-wallet/src/wallet/store.ts

```diff
@@ -391,14 +400,37 @@ export class Store {
         return this.ensureOpenChannelObjective(objective, tx);
       case 'CloseChannel':
         return this.ensureCloseChannelObjective(objective, tx);
-
       case 'SubmitChallenge':
         return this.ensureSubmitChallengeObjective(objective, tx);
+      case 'DefundChannel':
+        return this.ensureDefundChannelObjective(objective, tx);
       default:
         throw new StoreError(StoreError.reasons.unimplementedObjective);
     }
   }
 
+  private async ensureDefundChannelObjective(
+    objective: DefundChannel,
+    tx: Transaction
+  ): Promise<DBObjective> {
+    const {data} = objective;
+    const {targetChannelId} = data;
+    // fetch the channel to make sure the channel exists
+    const channel = await this.getChannelState(targetChannelId, tx);
+    if (!channel) {
+      throw new StoreError(StoreError.reasons.channelMissing, {channelId: targetChannelId});
+    }
```

Comment:
Shouldn't there be a foreign key constraint that guarantees this?

Reply:
The `data` of an `objective` does not necessarily have a `channelId`. It's also just `jsonb` column so it seems like this would be hard to encode as a foreign key.

Reply:
I am unfortunately unfamiliar with the current schema. My gut thought, when I wrote this, that there ought to be built in protection against there being no channel for an objective, if it's important for the objective to be attached to a channel.

Reply:
I think this probably makes sense but it would probably require changing the current objective schema a bit and I don't think it's worth the pain right now?

Reply:
Yep!